core/src/test/scala/unit/kafka/network/RequestConvertToJsonTest.scala

```diff
@@ -0,0 +1,140 @@
+
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.message._
+import org.junit.Test
+import org.apache.kafka.common.protocol.{ApiKeys, Errors}
+import org.apache.kafka.common.record.{CompressionType, MemoryRecords, RecordBatch, TimestampType}
+import org.apache.kafka.common.requests.OffsetsForLeaderEpochRequest.PartitionData
+import org.apache.kafka.common.requests._
+
+import scala.collection.mutable.ArrayBuffer
+
+class RequestConvertToJsonTest {
+
+  @Test
+  def testAllRequestTypesHandled(): Unit = {
+    val unhandledKeys = ArrayBuffer[String]()
+    ApiKeys.values().foreach(key => {
+      val version: Short = 0
+      var req: AbstractRequest = null
+      if (key == ApiKeys.PRODUCE) {
+        // There's inconsistency with the toStruct schema in ProduceRequest
+        // and ProduceRequestDataJsonConverters where the field names don't
+        // match so the struct does not have the correct field names. This is
+        // a temporary workaround until ProduceRequest starts using ProduceRequestData
+        req = ProduceRequest.Builder.forCurrentMagic(0.toShort, 10000, new util.HashMap[TopicPartition, MemoryRecords]()).build()
+      } else {
+        val struct = ApiMessageType.fromApiKey(key.id).newRequest().toStruct(version)
+        req = AbstractRequest.parseRequest(key, version, struct)
+      }
+      try {
+        RequestConvertToJson.request(req, false)
+      } catch {
+        case _ : AssertionError => unhandledKeys += key.toString
+      }
+    })
+    assert(unhandledKeys.isEmpty, String.format("%s request keys not handled in RequestConvertToJson", unhandledKeys))
+  }
+
+  @Test
+  def testAllResponseTypesHandled(): Unit = {
+    val unhandledKeys = ArrayBuffer[String]()
+    ApiKeys.values().foreach(key => {
+      val version: Short = 0
+      val struct = ApiMessageType.fromApiKey(key.id).newResponse().toStruct(version)
+      val res = AbstractResponse.parseResponse(key, struct, version)
+      try {
+        RequestConvertToJson.response(res, version)
+      } catch {
+        case _ : AssertionError => unhandledKeys += key.toString
+      }
+    })
+    assert(unhandledKeys.isEmpty, String.format("%s response keys not handled in RequestConvertToJson", unhandledKeys))
+  }
+
+  @Test
+  def testFormatOfOffsetsForLeaderEpochRequestNode(): Unit = {
+    val partitionDataMap = new util.HashMap[TopicPartition, PartitionData]
+    partitionDataMap.put(new TopicPartition("topic1", 0), new PartitionData(Optional.of(0),  1))
+
+    val version: Short = 3
+    val request = OffsetsForLeaderEpochRequest.Builder.forConsumer(partitionDataMap).build(version)
+    val manualGenNode = RequestConvertToJson.request(request, false)
+
+    val requestData = OffsetForLeaderEpochRequestDataJsonConverter.read(manualGenNode, version)
+    val autoGenNode = OffsetForLeaderEpochRequestDataJsonConverter.write(requestData, version)
+
+    assert(manualGenNode.equals(autoGenNode))
```

Comment:
```suggestion
    assertEquals(manualGenNode, autoGenNode)
```
using `assertEquals` instead of assert(`boolean`) makes this test easier to debug, since if it fails we get a print out of the objects. ditto for other `testFormat` tests.